src/utils/fs.spec.js

```diff
@@ -1,39 +1,189 @@
+const fs = require('fs')
+const path = require('path')
+
+jest.mock('fs')
+jest.mock('path')
+jest.mock('util', () => ({
+  promisify: jest.fn().mockImplementation(fn => fn),
+}))
+
+const toPathObject = (directory, filename) => ({
+  dir: directory,
+  base: filename,
+  root: '/',
+})
 
 describe('access()', () => {
-  it.todo('Calls fs.access')
+  it('Calls fs.access', () => {
+    const { access } = jest.requireActual('./fs')
+    access()
+
+    expect(fs.access).toHaveBeenCalledTimes(1)
+  })
 })
 
 describe('lstat()', () => {
-  it.todo('Calls fs.lstat')
+  it('Calls fs.lstat', () => {
+    const { lstat } = jest.requireActual('./fs')
+
+    lstat()
+
+    expect(fs.lstat).toHaveBeenCalledTimes(1)
+  })
 })
 
 describe('mkdir()', () => {
-  it.todo('Calls fs.mkdir')
+  it('Calls fs.readdir', () => {
+    const { mkdir } = jest.requireActual('./fs')
+
+    mkdir()
+
+    expect(fs.mkdir).toHaveBeenCalledTimes(1)
+  })
 })
 
 describe('readDir()', () => {
-  it.todo('Calls fs.readdir')
+  it('Calls fs.readdir', () => {
+    const { readDir } = jest.requireActual('./fs')
+
+    readDir()
+
+    expect(fs.readdir).toHaveBeenCalledTimes(1)
+  })
 })
 
 describe('readFile()', () => {
-  it.todo('Calls fs.readFile')
+  it('Calls fs.readFile', () => {
+    const { readFile } = jest.requireActual('./fs')
+
+    readFile()
+
+    expect(fs.readFile).toHaveBeenCalledTimes(1)
+  })
 })
 
 describe('stat()', () => {
-  it.todo('Calls fs.stat')
+  it('Calls fs.stat', () => {
+    const { stat } = jest.requireActual('./fs')
+
+    stat()
+
+    expect(fs.stat).toHaveBeenCalledTimes(1)
+  })
 })
 
 describe('writeFile()', () => {
-  it.todo('Calls fs.writeFile')
+  it('Calls fs.writeFile', () => {
+    const { writeFile } = jest.requireActual('./fs')
+
+    writeFile()
+
+    expect(fs.writeFile).toHaveBeenCalledTimes(1)
+  })
 })
 
 describe('findUpwardsFile()', () => {
-  it.todo('looks for file in the supplied directory')
-  it.todo('defaults to process.cwd()')
-  it.todo('returns an absolute path when matching file is found')
-  it.todo('returns false when file cannot be found')
-  it.todo('traverses the directory tree')
-  it.todo('???')
+  const directory = '/home/test'
+  const filename = 'test.file'
+  const parsedPath = toPathObject(directory, filename)
+  const errors = {
+    random: {
+      code: 'TEST',
+    },
+    enoent: {
+      code: 'ENOENT',
+    },
+  }
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+
+    path.parse.mockReturnValue(parsedPath)
+    path.join.mockReturnValue(directory.concat('/', filename))
+  })
+  it('looks for file in the supplied directory', async () => {
+    const { findUpwardsFile } = jest.requireActual('./fs')
+
+    path.parse.mockReturnValue(parsedPath)
+    path.join.mockReturnValue(directory.concat('/', filename))
```

Comment:
I think for the mostpart the path module doesn't have to be mocked at all. Ideally as few things as possible should be mocked to avoid testing a unit by it's code rather than it's logic.

The path module does not actually interact with the file system, so the portability of the tests shouldn't be affected (...I think? :D) by scrapping our tests dependency on it.

Reply:
Makes sense, I will stop mocking path.